### PR TITLE
fix: absolute url for windows

### LINKED ISSUE #650: Illegal character "\\" on windows while generating path for parquet files?
Screenshot 2024-02-06 090110

Some unknow trace error

```
thread '<unnamed>' panicked at server\src\storage\localfs.rs:217:10:
called `Result::unwrap()` on an `Err` value: BadSegment { path: "D:\\dev\\pinit\\data\\teststream/date=2024-02-06/hour=03/minute=28/Loralie.data.parquet", source: InvalidPart { segment: "D:\\dev\\pinit\\data\\teststream", illegal: "\\" } }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at server\src\storage\localfs.rs:217:10:
called `Result::unwrap()` on an `Err` value: BadSegment { path: "D:\\dev\\pinit\\data\\teststream/date=2024-02-06/hour=03/minute=28/Loralie.data.parquet", source: InvalidPart { segment: "D:\\dev\\pinit\\data\\teststream", illegal: "\\" } }
thread '<unnamed>' panicked at server\src\storage\localfs.rs:217:10:
called `Result::unwrap()` on an `Err` value: BadSegment { path: "D:\\dev\\pinit\\data\\teststream/date=2024-02-06/hour=03/minute=28/Loralie.data.parquet", source: InvalidPart { segment: "D:\\dev\\pinit\\data\\teststream", illegal: "\\" } }
thread '<unnamed>' panicked at server\src\storage\localfs.rs:217:10:
called `Result::unwrap()` on an `Err` value: BadSegment { path: "D:\\dev\\pinit\\data\\teststream/date=2024-02-06/hour=03/minute=28/Loralie.data.parquet", source: InvalidPart { segment: "D:\\dev\\pinit\\data\\teststream", illegal: "\\" } }
```

### PR DESCRIPTION
**Pull request by Delta456**
Fixes https://github.com/parseablehq/parseable/issues/650

### Description

Fixes #650. 

~Uses `RelativePath::new` function for platform independent relative paths.~

EDIT: Uses `std::path::MAIN_SEPARATOR` instead of hardcore `/` which handles slashes accordingly. 

<hr>

This PR has:
- [x] been tested to ensure log ingestion and log query works.
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added documentation for new or modified features or behaviors.

### CODE REVIEW

**Comment by Delta456** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Delta456** [comment]
recheck

**Comment by Delta456** [comment]
PS: This is my first time writing Rust code so would love code review.

File: server/src/storage/localfs.rs [hunk 1]
```diff
@@ -211,10 +211,8 @@ impl ObjectStorage for LocalFS {
     }
 
     fn absolute_url(&self, prefix: &RelativePath) -> object_store::path::Path {
-        object_store::path::Path::parse(format!(
-            "{}",
-            self.root.join(RelativePath::new(prefix).as_str()).display()
-        ))
+        object_store::path::Path::parse(format!("{}", self.root.join(prefix.as_str()).display()).trim_start_matches(std::path::MAIN_SEPARATOR),
```
Comment (Delta456) [inline server/src/storage/localfs.rs]: I rechecked and `std::path::MAIN_SEPARATOR` seems like a better solution here.

File: server/src/storage/localfs.rs [hunk 2]
```diff
@@ -212,7 +212,8 @@ impl ObjectStorage for LocalFS {
 
     fn absolute_url(&self, prefix: &RelativePath) -> object_store::path::Path {
         object_store::path::Path::parse(
-            format!("{}", self.root.join(prefix.as_str()).display()).trim_start_matches('/'),
+            format!("{}", self.root.join(prefix.as_str()).display())
+                .trim_start_matches(std::path::MAIN_SEPARATOR),
```
Comment (Delta456) [inline server/src/storage/localfs.rs:216]: Using `std::path::MAIN_SEPARATOR` is a better solution as it handles slashes accordingly.

**Comment by nitisht** [comment]
@Delta456 do you have windows machine / have you tested this ?

**Comment by Delta456** [comment]
> @Delta456 do you have windows machine / have you tested this ?

Yes I have this on my Windows machine.

image

**Review by nitisht — APPROVED** [review]

**Pull request merged**